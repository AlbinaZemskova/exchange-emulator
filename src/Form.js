import React, { useState } from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';


const Form = ({ handleSubmit }) => {
  const [price, setPrice] = useState('');
  const [count, setCount] = useState('');

  return (
    <form
      className="mt-3"
      onSubmit={e => handleSubmit(price, count, e)}
    >
      <span>
        Count of bitcoin
      </span>
      <br />
      <input
        value={count}
        className="w-50"
        onChange={e => setCount(+e.target.value)}
        id="count"
        autoComplete="off"
      />
      <br />
      <span>
        Price of 1 bitcoin
      </span>
      <br />
      <input
        value={price}
        className="w-50"
        onChange={e => setPrice(+e.target.value)}
        id="price"
        autoComplete="off"
      />
      <br />
      <Button
        className="mt-2"
        color="success"
        type="submit"
      >
        Submit
      </Button>
    </form>
  );
};

Form.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default Form;
