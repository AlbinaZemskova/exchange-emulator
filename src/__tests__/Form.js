import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import Form from '../Form';
import 'jest-prop-type-error';

Enzyme.configure({ adapter: new Adapter() });

describe('<Form />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Form handleSubmit={jest.fn()} />,
    );
  });

  describe('render', () => {
    it('renders without crashing', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('checking proptypes', () => {
      expect(() => (
        <Form handleSubmit={{}} />
      )).toThrow();
    });

    describe('method', () => {
      it('changed price filed', () => {
        const event = {
          target: {
            value: '1000',
            id: 'price',
          },
        };

        expect(wrapper.state().price).toBe('');
        wrapper.instance().onFieldChange(event);
        expect(wrapper.state()[event.target.id]).toEqual(+event.target.value);
      });

      it('changed count filed', () => {
        const event = {
          target: {
            value: '100',
            id: 'count',
          },
        };

        expect(wrapper.state().count).toBe('');
        wrapper.instance().onFieldChange(event);
        expect(wrapper.state()[event.target.id]).toEqual(+event.target.value);
      });
    });
  });
});
