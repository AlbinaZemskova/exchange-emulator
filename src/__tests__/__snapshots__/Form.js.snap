// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> render renders without crashing 1`] = `
<form
  className="mt-3"
  onSubmit={[Function]}
>
  <span>
    Count of bitcoin
  </span>
  <br />
  <input
    autoComplete="off"
    className="w-50"
    id="count"
    onChange={[Function]}
  />
  <br />
  <span>
    Price of 1 bitcoin
  </span>
  <br />
  <input
    autoComplete="off"
    className="w-50"
    id="price"
    onChange={[Function]}
  />
  <br />
  <Button
    className="mt-2"
    color="success"
    tag="button"
    type="submit"
  >
    Submit
  </Button>
</form>
`;
